src/kernel/arch/aarch64/mmio.zig

```diff
@@ -1,31 +1,87 @@
-const GPIO_OFFSET: usize = 0x200000;
-const UART_OFFSET: usize = GPIO_OFFSET + 0x1000;
+const log = @import("../../log.zig");
+const std = @import("std");
+
+const MMIO_OFFSET: usize = 0x200000;
+const GPIO_OFFSET: usize = MMIO_OFFSET + 0x0000;
+const UART_OFFSET: usize = MMIO_OFFSET + 0x1000;
+const AUX_OFFSET: usize = MMIO_OFFSET + 0x15000;
 const MBOX_OFFSET: usize = 0xB880;
 
 pub const Register = enum(usize) {
-    GPIO_PULL = GPIO_OFFSET + 0x94,
-    GPIO_PULL_CLK = GPIO_OFFSET + 0x98,
-    UART_DATA = UART_OFFSET + 0x00,
-    UART_FLAGS = UART_OFFSET + 0x18,
-    UART_CONTROL = UART_OFFSET + 0x30,
-    UART_INT_CONTROL = UART_OFFSET + 0x44,
-    UART_BAUD_INT = UART_OFFSET + 0x24,
-    UART_BAUD_FRAC = UART_OFFSET + 0x28,
-    UART_LINE_CONTROL = UART_OFFSET + 0x2C,
-    UART_INT_MASK = UART_OFFSET + 0x38,
+    AUX_IRQ = AUX_OFFSET + 0x00,
+    AUX_ENABLES = AUX_OFFSET + 0x04,
+    AUX_MU_IO_REG = AUX_OFFSET + 0x40,
+    AUX_MU_IER_REG = AUX_OFFSET + 0x44,
+    AUX_MU_IIR_REG = AUX_OFFSET + 0x48,
+    AUX_MU_LCR_REG = AUX_OFFSET + 0x4c,
+    AUX_MU_MCR_REG = AUX_OFFSET + 0x50,
+    AUX_MU_LSR_REG = AUX_OFFSET + 0x54,
+    AUX_MU_MSR_REG = AUX_OFFSET + 0x58,
+    AUX_MU_SCRATCH = AUX_OFFSET + 0x5c,
+    AUX_MU_CNTL_REG = AUX_OFFSET + 0x60,
+    AUX_MU_STAT_REG = AUX_OFFSET + 0x64,
+    AUX_MU_BAUD_REG = AUX_OFFSET + 0x68,
+    GPIO_FSEL0 = GPIO_OFFSET + 0x00,
+    GPIO_FSEL1 = GPIO_OFFSET + 0x04,
+    GPIO_FSEL2 = GPIO_OFFSET + 0x08,
+    GPIO_FSEL3 = GPIO_OFFSET + 0x0c,
+    GPIO_FSEL4 = GPIO_OFFSET + 0x10,
+    GPIO_FSEL5 = GPIO_OFFSET + 0x14,
+    GPIO_SET0 = GPIO_OFFSET + 0x1c,
+    GPIO_SET1 = GPIO_OFFSET + 0x20,
+    GPIO_CLR0 = GPIO_OFFSET + 0x28,
+    GPIO_CLR1 = GPIO_OFFSET + 0x2c,
+    GPIO_PUD = GPIO_OFFSET + 0x94,
+    GPIO_PUDCLK0 = GPIO_OFFSET + 0x98,
+    GPIO_PUDCLK1 = GPIO_OFFSET + 0x9c,
     MBOX_STATUS = MBOX_OFFSET + 0x18,
     MBOX_WRITE = MBOX_OFFSET + 0x20,
     MBOX_READ = MBOX_OFFSET + 0x00,
+    UART_DR = UART_OFFSET + 0x00,
+    UART_FR = UART_OFFSET + 0x18,
```

Comment:
For the registers that existed before, could we keep old names? They're a bit more descriptive.

Reply:
The reference for these is https://www.raspberrypi.org/documentation/hardware/raspberrypi/bcm2835/BCM2835-ARM-Peripherals.pdf which also applies generally to the bcm2837 which is on the pi3b and pi3b+.

The names I used are from this manual but using your prefix (GP instead of GPIO.) What I would like to do is use the ones straight from the manual, unmodified. They are not as descriptive, but they are what everyone else is using.

Your decision.

Reply:
I see your reasoning. I still think that more descriptive names are better though. Perhaps we can convince the rest of the world too :)

Reply:
Ok, the hex offsets are an unambiguous link between the source and the manual, so more descriptive names will work.

Reply:
> For the registers that existed before, could we keep old names? They're a bit more descriptive.

Done: 0a034d3c956f98a70001b24ed4453ccb3520fe9d